securedrop/models.py

```diff
@@ -784,13 +790,57 @@ def to_json(self, all_info: bool = True) -> Dict[str, Any]:
 
         return json_user
 
+    def delete(self) -> None:
```

Comment:
I would recommend explicitly passing the db_session as an argument:

- It will make testing easier.
- Importing the db object/module directly creates the implicit assumption that the caller has setup the special (flask) context where this import will work. Making the db_session an argument makes this requirement explicit/documented, and leaves it to the caller to get the db session.

Reply:
All the other methods in Journalist don't do this, so I'd prefer to keep it consistent in the class for now. What you explained seems reasonable to me, I just think it should be done separately.